ts_src/transaction_builder.ts

```diff
@@ -197,13 +236,36 @@ export class TransactionBuilder {
   }
 
   sign(
-    vin: number,
-    keyPair: ECPairInterface,
+    signParams: number | TxbSignArg,
+    keyPair?: ECPairInterface,
     redeemScript?: Buffer,
     hashType?: number,
     witnessValue?: number,
     witnessScript?: Buffer,
   ): void {
+    let vin: number;
+    if (typeof signParams === 'number') {
+      console.warn(
+        'DEPRECATED: TransactionBuilder sign method arguments ' +
+          'will change in v6, please use the TxbSignArg interface',
+      );
+      vin = signParams;
+    } else if (typeof signParams === 'object') {
+      checkSignArgs(this, signParams);
+      vin = signParams.vin;
+      keyPair = signParams.keyPair;
+      redeemScript = signParams.redeemScript;
+      hashType = signParams.hashType;
+      witnessValue = signParams.witnessValue;
+      witnessScript = signParams.witnessScript;
```

Comment:
This block:

```js
keyPair = signParams.keyPair;
redeemScript = signParams.redeemScript;
hashType = signParams.hashType;
witnessValue = signParams.witnessValue;
witnessScript = signParams.witnessScript;
```

Could be written as:

```js
({
    keyPair,
    redeemScript,
    hashType,
    witnessValue,
    witnessScript,
} = signParams);
```

to reduce code duplication.

Reply:
I thought that only worked with let or const, and these were already declared... let me try it out and if it works I'll push it

Reply:
yup, it worked. TIL! Thanks.

Reply:
It works with with predefined variables if you wrap it in brackets.

e.g:

No brackets for declaring new variables:

```js
const {
	vin,
	keyPair,
	redeemScript,
	hashType,
	witnessValue,
	witnessScript,
} = signParams;
```

Variables are predefined so we need to wrap the expression in brackets.

```js
let vin;
let keyPair;
let redeemScript;
let hashType;
let witnessValue;
let witnessScript;

({
	vin,
	keyPair,
	redeemScript,
	hashType,
	witnessValue,
	witnessScript,
} = signParams);
```